src/ICSharpCode.SharpZipLib/Zip/ZipFile.cs

```diff
@@ -4038,11 +4038,27 @@ public override int Read(byte[] buffer, int offset, int count)
 						baseStream_.Seek(readPos_, SeekOrigin.Begin);
 					}
 					int readCount = baseStream_.Read(buffer, offset, count);
+
+					// Azure CloudBlockBlob SDK Stream implementation is: Returns only the bytes in its buffer and does not fetch from remote until its buffer is empty.
+					// This observation causes an issue in above baseStream_.Read call because the buffer is only filled partially
+					int correctReadCount = readCount;
```

Comment:
This file works according to spec. If the underlying stream returns fewer bytes, it correctly returns the same amount of bytes. The changes does not belong here (and neither does the comment about third-party specific details).

Reply:
I agree that the change can be applied in the caller ZipAESStream.cs file too. Thanks for looking and let me know if you applied that fix and I will help test. You can ignore this PR though.